src/ComputeSharp.D2D1/Shaders/Interop/D2D1ResourceTextureManager.cs

```diff
@@ -0,0 +1,487 @@
+///         [in, optional] const UINT32                           *strides,
+///                        UINT32                                 dataSize);
+/// 
+///     HRESULT Update(
+///         [in, optional] const UINT32 *minimumExtents,
+///         [in, optional] const UINT32 *maximimumExtents,
+///         [in]           const UINT32 *strides,
+///                        UINT32       dimensions,
+///         [in]           const BYTE   *data,
+///                        UINT32       dataCount);
+/// };
+/// </code>
+/// </para>
+/// <para>
+/// These can be used to create resource textures, pass them to effects, and also share resources among different effects.
+/// </para>
+/// <para>
+/// For details of how these work, <c>Initialize</c> maps to <c>ID2D1EffectContext::CreateResourceTexture</c>, whereas
+/// <c>Update</c> maps to <c>ID2D1ResourceTexture::Update</c>. The same behavior and parameters as those methods is used.
+/// </para>
+/// <para>
+/// This interface is implemented by ComputeSharp.D2D1, and it can be used through the APIs in <see cref="D2D1ResourceTextureManager"/>.
+/// That is, <see cref="Create"/> is first use to create an <c>ID2D1ResourceTextureManager</c> instance. Then, <see cref="Initialize"/>
+/// can be used to initialize the resource texture held by the manager. The manager can also be assigned to an effect at any time,
+/// using the available property indices from <see cref="D2D1PixelShaderEffectProperty"/>. To update texture data after its initial
+/// creation, <see cref="Update(void*, ReadOnlySpan{uint}, ReadOnlySpan{uint}, ReadOnlySpan{uint}, ReadOnlySpan{byte})"/> can be used.
+/// </para>
+/// <para>
+/// An RCW (runtime callable wrapper, see <see href="https://docs.microsoft.com/dotnet/standard/native-interop/runtime-callable-wrapper"/>),
+/// is also available for all of these APIs, implemented by the same <see cref="D2D1PixelShaderEffect"/>. In this case, the constructor can
+/// be used to create and initialize an instance (equivalent to calling <see cref="Create"/> and <see cref="Initialize"/>, and then
+/// <see cref="Update(ReadOnlySpan{uint}, ReadOnlySpan{uint}, ReadOnlySpan{uint}, ReadOnlySpan{byte})"/> is available for texture data updates.
+/// The instance implements <see cref="ICustomQueryInterface"/>, which can be used in advanced scenarios to retrieve the underlying COM object.
+/// </para>
+/// <para>
+/// The <c>ID2D1ResourceTextureManager</c> contract allows callers to create a resource texture at any time even before the effect
+/// has been initialized and the manager assigned to it. In that case, the manager will buffer the data internally, and will defer
+/// the creation of the actual resource texture until it's possible for it to do so, at which point it will also free the buffer.
+/// </para>
+/// <para>
+/// If implementing a custom resource manager was needed, in order for external implementations to be accepted by the effect objects
```

Comment:
proofreading nit: ... ~~was~~ is needed, ...

Reply:
Fixed in 28ed19c349592ecaafaafd00328df254e7556db9.